react/features/authentication/components/native/AuthenticationPage.tsx

```diff
@@ -0,0 +1,85 @@
+import { Route } from '@react-navigation/native';
+import React, { PureComponent } from 'react';
+import { WithTranslation } from 'react-i18next';
+import { View, ViewStyle } from 'react-native';
+import { WebView } from 'react-native-webview';
+import { connect } from 'react-redux';
+
+import { IStore } from '../../../app/types';
+import { translate } from '../../../base/i18n/functions';
+import JitsiScreen from '../../../base/modal/components/JitsiScreen';
+import LoadingIndicator from '../../../base/react/components/native/LoadingIndicator';
+
+import styles, { INDICATOR_COLOR } from './styles';
+
+
+interface IProps extends WithTranslation {
+
+    dispatch: IStore['dispatch'];
+
+    /**
+     * Default prop for navigating between screen components(React Navigation).
+     */
+    navigation: any;
+
+    /**
+     * Default prop for navigating between screen components(React Navigation).
+     */
+    route: Route<'', { tokenAuthServiceUrl: string; }>;
+}
+
+/**
+ * Implements a React native component that displays the authentication page for a specific room.
+ */
+class AuthenticationPage extends PureComponent<IProps> {
```

Comment:
Doesn't it make sense to use a functional component here?

Reply:
Converted.